packages/nimble-components/src/table/specs/data-hierarchy-hld.md

```diff
@@ -0,0 +1,168 @@
+# Data Hierarchy
+
+## Problem Statement
+
+There are scenarios where clients want to provide collapsible hierarchy to the rows being presented in the table, where both parent and children rows provide values in typically at least a subset of the same columns.
+
+For example, consider a table of geographic information that included location, population, flag, and possibly other location-related information. The location column could be either a continent, country, state, or any sub-regional construct. It would be reasonable to want to present this information in a way that allowed a user to collapse/expand any outer regional grouping, which itself would contain all rows whose locations existed inside it (e.g. 'United States' would be parent of all rows whose location was a U.S. state).
+
+### Lazy Loading
+
+In addition to just supporting data hierarchy with a fully provided set of data, we also need to support the scenario where a row is presented as a parent, but none of its children are currently represented in the data. Upon expanding the row, the client is then expected to load all of the data for that parent and provide it to the table (via the standard `setData` method).
+
+## Links To Relevant Work Items and Reference Material
+
+-   [#890: Design for hierarchical data support in tables](https://github.com/ni/nimble/issues/890)
+-   [#861: Table hierarchical data support](https://github.com/ni/nimble/issues/861)
+-   [Prototype using flat list that includes a parentId](https://github.com/ni/nimble/tree/data-hierarchy-flat-list-prototype) with [Storybook](https://60e89457a987cf003efc0a5b-qzwoshcidz.chromatic.com/?path=/story/incubating-table--table&args=data:LargeDataSet)
+-   [Prototype with user data being hierarchical](https://github.com/ni/nimble/tree/data-hierarchy-prototype) with [Storybook](https://60e89457a987cf003efc0a5b-yncupvnoes.chromatic.com/?path=/story/incubating-table--table&args=data:LargeDataSet)
+
+## Implementation / Design
```

Comment:
It would be good to capture our plans for interactions:
1. collapse all button is updated to collapse hierarchical rows in addition to group rows
2. what is the mouse interaction to expand/collapse a row: click anywhere on the row or only on certain parts of the row? If you can click anywhere, would a selectable row still become selected when you click on it?
3. what are the expected keyboard interactions for expand/collapse? (or add this as a decision / open issue on #1506)